### DISCUSSION TITLE
wo sind die Einstellungen, die diese Version vom Original unterscheidet?

### DISCUSSION THREAD

**Discussion by peff74** [hoylabs/OpenDTU-OnBattery#186] [Q&A] [answered]
Hi,

ich habe jetzt schon einige Mal verzweifelt versucht diese onBatterie Version auf meinen ESP32 zu bringen.
Genutzt habe ich dafür die 4 Dateien die unter den Releases im .zip liegen mit der Version 23.4.7.

Aber nach der Installation sieht die Weboberfläche identisch zur Originalen aus?!?

Was mache ich falsch?
Help!

**Comment by helgeerbe** [comment]
@peff74 wahrscheinlich gar nichts. Hast du denn den mal den Cache vom Browser aktualisiert. f5, reload oder was immer man da für deinen Browser so braucht.

**Comment by bf4ed** [comment]
Währe aber gut das auf den ersten Blick zu erkennen. Einfach "on Battery" dazu zu schreiben in den  Titel. Du erkennst es erst wenn du auf Einstellungen gehst.

**Reply by helgeerbe** [reply]
![image](https://user-images.githubusercontent.com/59169507/234555500-113e6a38-ba57-414e-a97b-8110d05fb8de.png)

Habe jetzt ein Batterysymbol dazugelegt.

**Comment by madmartin** [comment]
Da hat @bf4ed total recht.....

**Comment by peff74** [comment]
Also F5 und refresh bringen mir auch nix zu Tage.
Bin jetzt noch mal die ganzen Einstellungen druchgegangen... da is nigends der gleichen zu finden.
![2023-04-26 11_33_33-helgeerbe_OpenDTU-OnBattery_ Software for ESP32 to talk to Hoymiles Inverters an](https://user-images.githubusercontent.com/91614910/234534959-8082e244-0d41-41c7-b02f-329e988f527a.png)

Das hier ist mein Menu.
![Screenshot_20230426_113504_Chrome](https://user-images.githubusercontent.com/91614910/234535824-438d1138-2c09-4490-b3bd-11a01e7f5560.jpg)

So schwer kann das doch nicht sein.
Ich kenne OpenDTU bisher nicht hatte Ahoy, aber so blind kann ich ja hoffentlich nicht sein, oder?

**Comment by madmartin** [comment]
@peff74 Im Einstellungs-Menü müssten mehr Menüpunkte sein.

Letzte Chance: Bitte mal den Browser-Cache löschen.
Die "force reload" Funktion bei den verschiedenen Browser ist unterschiedlich implementiert, F5 plus Shift oder STRG oder sonstwas....

**Comment by peff74** [comment]
dachte ich es mir... dass es dort weitere Menüpunkte geben muss.
Cache scheidet aus. Schon unterschiedliche Devices genutzt.

Und zur Installation nehme ich diese Quelle, korrekt?
![2023-04-26 13_13_17-Releases · helgeerbe_OpenDTU-OnBattery](https://user-images.githubusercontent.com/91614910/234562126-894180a1-687a-4f5c-86be-c27ab820eaa3.png)

Und auf den ESP schiebe ich es so wie openDTU auch, oder?
![2023-04-26 13_33_11-Releases · helgeerbe_OpenDTU-OnBattery](https://user-images.githubusercontent.com/91614910/234562684-2b0a98d2-1cac-4861-8f5a-bab1f097d0e7.png)

Das gibt's doch gar net, verdammte Axt.
Ich teste das gleich noch einmal.

**Reply by madmartin** [reply]
Achtung, ich glaube das Bild vom "Flash Download Tool" enthält nicht die aktuellen Adressen! Bitte mit dem README.md vergleichen und im Zweifel die Zieladressen aus dem README.md verwenden!

Und die Frage "wie OpenDTU auch" -> ja! von dort kommt ja der "Breaking Change" mit der geänderten Einteilung des Flash

**Reply by peff74** [reply]
Also die Speicheradressen sind alle gleich, oder ich habe einen auf'm Auge.
![2023-04-26 15_00_48-OpenDTU-OnBattery_README md at master · helgeerbe_OpenDTU-OnBattery](https://user-images.githubusercontent.com/91614910/234583092-c0e9b6f2-a0d7-40da-ac54-68ade763fcef.png)

**Comment by swingstate** [comment]
Wenn ich die OpenDTU-generic files aus dem Release link flashe landete ich bisher auch immer bei diesem Ergebnis. Ich bin dazu übergegangen den development branch via StudioCode zu kompilieren und zu flashen. 

Nicht unbedingt ein Ansatz für jeder Man, aber funktioniert wenigstens.

**Reply by swingstate** [reply]
Für den Anwender wäre es wohl auch einfacher, wenn aus dem Dateinamen hervor geht, dass es sich nicht um OpenDTU-ohne-extras-und-battery handelt, sondern eben um OpenDTU-onBattery.

**Comment by peff74** [comment]
O.K. doch nicht zu blöde.
Wobei man darauf achten muss, dass der Link auf onBatterie zu den "normlen" Releases führt. 
![2023-04-26 13_43_07-Posteingang - stefan meier@dssmith com - Outlook](https://user-images.githubusercontent.com/91614910/234565169-2734e413-66b9-4f97-98ec-d7df9f666023.png)

Also nutzt Du PlatformIO?
Das habe ich mal versucht, bin aber ganz schnell wieder zur Arduino IDE zurückgelaufen^^.
Oder gibt es da einen einfacheren Weg?

**Reply by swingstate** [reply]
Ja. Ich bin nach [dieser Anleitung](https://github.com/helgeerbe/OpenDTU-OnBattery#flashing-and-starting-up) vorgegangen, aber Achtung auch dort wird auf das Originalprojekt verwiesen. 
Ich meine gesehen zu haben, dass an der Readme gearbeitet wird, da diese ziemlich outdated und teilweise umkonkret ist.

**Reply by madmartin** [reply]
Oh tatsächlich, der Link in der README.md von ---onBattery führt ins originale OpenDTU - was allein für sich falsch ist.
Der ganze Abschnitt muß angepasst werden,  denn es gibt ja jetzt die GitHub Releases, die viel einfacher zu erreichen sind:
![grafik](https://user-images.githubusercontent.com/4208120/234567276-02fa513f-caff-4949-9eaa-bed590213db2.png)

Ich mach heut abend mal einen entsprechenden PR bei OpenDTU rein um das zu ändern.

**Reply by madmartin** [reply]
>  Ich meine gesehen zu haben, dass an der Readme gearbeitet wird, da diese ziemlich outdated und teilweise umkonkret ist.


 Du meinst bestimmt #182 aber es passt immer noch nicht.... in beiden Readme's

**Reply by swingstate** [reply]
Genau, Zeile 108 der Readme in verweist dort noch auf "OpenDTU". Ich werde mal ein paar Kurse bei der YouTube Universität zum Thema PRs/Arbeiten mit und auf Github belegen. Dann kann ich vielleicht auch etwas mehr beitragen. :)

**Comment by helgeerbe** [comment]
Über Hilfe bei der Dokumentation würde ich mich sehr freuen. Da alles kostet doch viel Zeit und es wäre gut, wenn man das auf mehrere Schultern verteilen würde.

**Comment by peff74** [comment]
Gibt es jetzt eine Chance aus dem Release nun über das ESP Tool eine openDTU onBatt auf den ESP zu bekommen?
Werden die irgendwie automatisch in GIT generiert oder macht Ihr das?
Vielleicht ist da was falsch verlinkt, wenn es automatisch geht, sodass es auch auf das Original verweist und daraus das die BIN aus dem Original generiert wird?

**Reply by helgeerbe** [reply]
Ich habe das mir eben mal angeschaut. Habe jetzt auf die Schnell unter actions nichts gesehen, was auf das original Repo verweist. Habe zumindest im Development Branch schon mal die Readme angepasst: 
![image](https://user-images.githubusercontent.com/59169507/234593074-d6f04408-cba8-4ac2-aacc-640dd94fd9b2.png)

Klick drauf, sollte dich zu den Actions in Git führen. Und in der Build Action sollten sich die Binaries befinden.

**Comment by peff74** [comment]
Auch das hat nicht funktioniert...
Kam wieder die Version dabei raus ohne 'onBattery'

Also habe ich mich durch VS Code über Platformio zum Gitclone und eeeeewig langem kompilieren gequält.
Und siehe da. Ich habe nun eine Version dort befindet sich ein Akku Symbol hinter dem OpenDTU und die Einstellung beinhalten alles was das Herz begehrt.
Welche Version ich da jetzt allerdings erwischt habe, weiß ich nicht :-(
Es steht dort nur ein Hash--> 83b42a3. Und wenn ich auf den Link klicke, lande ich wieder beim 'normalen' OpenDTU.

Alles irgendwie für mich nicht verständlich.

**Comment by peff74** [comment]
Habe jetzt herausgefunden, dass die Hash das 'Master' von 'OnBattery' ist.
Testhalber habe ich nun auch über VS Code die v23.4.25  kompeliert.
Und was soll ich sagen, siehe da --> Nix is mehr mit OnBattery Funktionen!!!

![2023-04-26 18_50_57-](https://user-images.githubusercontent.com/91614910/234648328-9a9c8675-4da5-4a9b-b55f-36cc08053954.png)

Irgendwas stimmt da nicht, oder ich verstehe es nicht.
Habe eh respekt vor denen die sich mit GIT so richtig auskennen

**Reply by madmartin** [reply]
@peff74 auch für Dich: siehe weiter unten meine Antwort an Mogdar-M  Du kannst jetzt fertige Images aus dem Menü "Actions" runterladen.
Link 

https://github.com/helgeerbe/OpenDTU-OnBattery/actions/runs/4822527948

für das erste funktionierende Build.
Später wird es sicher von @helgeerbe "Releases" geben.

**Comment by madmartin** [comment]
Um die README.md des Hauptprojekts zu fixen habe ich pull-request https://github.com/tbnobody/OpenDTU/pull/851 platziert. @helgeerbe wird das dann automatisch hierher übernehmen.

**Comment by madmartin** [comment]
@helgeerbe  Ich habe jetzt mal ganz explizit in DIESEM Projekt das binary file runtergeladen und geflasht - und tatsächlich - es fehlen die Menüpunkte, die -onBattery ausmachen.

Mit dem Release-Build stimmt was nicht. Du hast zwar mit commit https://github.com/helgeerbe/OpenDTU-OnBattery/commit/e65b2196bfa5d91add0041cde5fd80b2d03de72d die modifizierten webapp committet, der Release scheint aber - zumindest nach Auskunft der Webseite https://github.com/helgeerbe/OpenDTU-OnBattery/releases  - auf commit 0441bbb zu basieden. Das ist derselbe commit wie auf https://github.com/tbnobody/OpenDTU/releases

**Reply by peff74** [reply]
Danke das Du hilfst, ich bin sicher nicht der Einzige der hier Stunden verzweifelt investierst

**Reply by madmartin** [reply]
Ja, das tut mir sehr leid, ich kann das nachvollziehen!

**Reply by swingstate** [reply]
Ich kann das Problem bestätigen, weshalb ich von den .bin Dateien weg bin und über VSCode flashe. Wenn ihr das fixt, @helgeerbe & @madmartin, könnt ihr dann vielleicht auch den Dateinamen der .bin auf OpenDTU-onBatter.bin oder ähnliches anpassen? Oder gibt das Konflikte?

**Reply by madmartin** [reply]
Ich habe noch nie was mit den github builds gemacht, aber ich denke das sollte möglich sein. Ich schaue es mir gerade an.
@helgeerbe ich sehe da zwei Möglichkeiten um dieses Dilemma anzustellen:
1. im automatischen Build auch die webapp frisch zu "builden" (EDIT: soweit ich das sehe wird bereits gemacht)
2. ein eigenes Release- und Tag Schema für dieses Sub-Projekt einzuführen, denn die "Releases" basieren ja auf Tags aus dem upstream OpenDTU - und wenn Du danach in einem Commit die Webapp neu machst brauchst Du auch ein neues, dem Sub-Projekt eigenes Tag. Vielleicht sowas wie das OpenDTU-Tag mit einem Suffix oder Prefix.

EDIT: Ich denke 2. wird die Lösung sein.

ADD: um andere Output-File Namen zu erhalten müssen in https://github.com/helgeerbe/OpenDTU-OnBattery/blob/master/.github/workflows/build.yml die Zeilen 99 und 142 angepasst werden.

**Reply by madmartin** [reply]
```
diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 4fa430e..294a6bc 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -96,16 +96,16 @@ jobs:
         run: pio run -e ${{ matrix.environment }}
 
       - name: Rename Firmware
-        run: mv .pio/build/${{ matrix.environment }}/firmware.bin .pio/build/${{ matrix.environment }}/opendtu-${{ matrix.environment }}.bin
+        run: mv .pio/build/${{ matrix.environment }}/firmware.bin .pio/build/${{ matrix.environment }}/opendtu-onbattery-${{ matrix.environment }}.bin
 
       - name: Copy boot_app0.bin
         run: cp ~/.platformio/packages/framework-arduinoespressif32/tools/partitions/boot_app0.bin .pio/build/${{ matrix.environment }}/boot_app0.bin
 
       - uses: actions/upload-artifact@v3
         with:
-          name: opendtu-${{ matrix.environment }}
+          name: opendtu-onbattery-${{ matrix.environment }}
           path: |
-            .pio/build/${{ matrix.environment }}/opendtu-${{ matrix.environment }}.bin
+            .pio/build/${{ matrix.environment }}/opendtu-onbattery-${{ matrix.environment }}.bin
             .pio/build/${{ matrix.environment }}/partitions.bin
             .pio/build/${{ matrix.environment }}/bootloader.bin
             .pio/build/${{ matrix.environment }}/boot_app0.bin
@@ -139,7 +139,7 @@ jobs:
           sudo apt install zip
           cd artifacts
           for i in */; do zip -r "${i%/}.zip" "$i"; done
-          for i in */; do cp ${i}opendtu-*.bin ./; done
+          for i in */; do cp ${i}opendtu-onbattery-*.bin ./; done
 
       - name: Create release
         uses: softprops/action-gh-release@v1
```

**Reply by swingstate** [reply]
müsste da jetzt nicht eine neue onbattery .bin zum Vorschein kommen? :)

**Reply by madmartin** [reply]
Nein, das sind nur meine Änderungs-Vorschläge, ich habe hier keine Berechtigung was zu committen....

**Reply by madmartin** [reply]
Committen - also was zu diesem Repository hinzufügen - kann @helgeerbe  und vielleicht noch @berni2288 wenn ich die History richtig deute....

**Comment by madmartin** [comment]
Es hat mir keine Ruhe gelassen: #192

**Comment by helgeerbe** [comment]
Danke für den Input. Ich habe die Actions tatsächlich nie benutzt und habe nur geschaut, dass auch die Builds und Lint-Prüfungen durchlaufen.

Das die Dokumentation ein Graus ist, ist mir durchaus bewusst. Nur allein mir fehlt die Zeit. Aktuell konzentriere ich mich noch auf Funktionalität. Und das ganze ist schon durch die tollen Beiträge von anderen mehr geworden als ich ursprünglich beabsichtigt hatte.

Aber ich kann immer nur wiederholen. Auch wenn man nicht programmieren kann. Wenn man Lücken in der Doku oder Fehler findet, kann man auch die Doku über Pull Requests verbessert. Man hilft dann allen zukünftigen Benutzern.

Wenn jemand die Oberhoheit über die Doku übernehmen möchte, würde ich mich freuen hier Unterstützung zu bekommen.

**Comment by madmartin** [comment]
Gerne.
Github macht es einem aber auch nicht leicht.... ich wollte mir einen Fork von "helgeerbe/OpenDTU-OnBattery" anlegen - sagt GH: "geht nicht, du hast schon einen Fork von tbnobody/OpenDTU-OnBattery". Fork vom Fork wird nicht unterstützt, steht explizit in der Doku, und keiner weiß warum. Git selbst hat damit kein Problem. Offizielle Workarounds: - eine zweite Userkennung anlegen / - den einen Fork löschen. dann den anderen anlegen. 
Ist beides Käse.

**Comment by Mogdar-M** [comment]
Ich habe gerade, nach zwei Tagen Kampf, openDTU onbattery geflasht
Allerdings sehe ich weder in der Version, die ich über VS Code geflasht habe noch nach dem OTA Update mit der bin auf 23.4.24 die Option, Zugriff auch weitere Optionen zu konfigurieren.

Mein Ziel ist ein Zugriff auf einen Victron, um die Daten via MQTT abzubilden

**Reply by madmartin** [reply, marked as answer]
Hallo @Mogdar-M ,
tut uns leid, da ist was mit den automatischen Builds schiefgelaufen.
Mittlerweile hat @helgeerbe das behoben. 
Du kannst jetzt unter diesem Link

https://github.com/helgeerbe/OpenDTU-OnBattery/actions/runs/4822527948

unten auf der Webseite bei "opendtu-onbattery-generic " ein Zip-Archiv runterladen.
Wenn openDTU schon auf deinem ESP läuft, kannst Du einfach das im Zip enthaltene opendtu-onbattery-generic.bin über die Web-Oberfläche updaten.

Nach dem reboot des ESP dann bitte unbedingt einen Reload der Weboberfläche erzwingen, damit die neuen Menüs auftauchen!

Have fun!
Martin

**Reply by Mogdar-M** [reply]
![image](https://user-images.githubusercontent.com/129499239/234961671-32f6f147-3152-4bc0-8f23-80c69e7066c4.png)
tada, hat funktioniert

Dann bau ich jetzt den Spannungsteiler an und die Verbindung zum Victron
